lib/chef/knife/vsphere_vm_clone.rb

```diff
@@ -345,23 +356,51 @@ def run
       unless config[:disable_customization]
         # Wait for customization to complete
         puts 'Waiting for customization to complete...'
-        CustomizationHelper.wait_for_sysprep(vm, vim, Integer(get_config(:sysprep_timeout)), 10)
+        CustomizationHelper.wait_for_sysprep(@vm, vim, Integer(get_config(:sysprep_timeout)), 10)
         puts 'Customization Complete'
-        sleep 2 until vm.guest.ipAddress
-        connect_host = config[:fqdn] = config[:fqdn] ? get_config(:fqdn) : vm.guest.ipAddress
+        sleep 2 until connect_host
+        connect_host
       end
-      wait_for_access(connect_host, connect_port, protocol)
+      wait_for_access(connect_port, protocol)
       ssh_override_winrm
       bootstrap_for_windows_node.run
     else
       protocol ||= 'ssh'
-      wait_for_access(connect_host, connect_port, protocol)
+      wait_for_access(connect_port, protocol)
       ssh_override_winrm
       bootstrap_for_node.run
     end
   end
 
-  def wait_for_access(connect_host, connect_port, protocol)
+  def ipv4_address
+    puts 'Waiting for Network Interfaces to become available...'
+    sleep 2 while @vm.guest.net.empty?
+    puts 'Waiting for a valid IPv4 address...'
+    # Multiple reboots occur during guest customization in which a link-local
+    # address is assigned. As such, we need to wait until a routable IP address
+    # becomes available. This is most commonly an issue with Windows instances.
+    sleep 2 while @vm.guest.net[config[:bootstrap_nic]].ipConfig.ipAddress.detect { |addr| IPAddr.new(addr.ipAddress).ipv4? }.origin == 'linklayer'
+    @vm.guest.net[config[:bootstrap_nic]].ipAddress.detect { |addr| IPAddr.new(addr).ipv4? }
```

Comment:
What's the difference between `@vm.guest.net[config[:bootstrap_nic]].ipConfig.ipAddress` and `@vm.guest.net[config[:bootstrap_nic]].ipAddress`?